src/proc/constant_evaluator.rs

```diff
@@ -604,15 +686,12 @@ impl<'a> ConstantSolver<'a> {
         op: BinaryOperator,
         left: Handle<Expression>,
         right: Handle<Expression>,
-        span: crate::Span,
-    ) -> Result<Handle<Expression>, ConstantSolvingError> {
-        let left = self.eval_zero_value(left, span)?;
-        let right = self.eval_zero_value(right, span)?;
-
-        let expr = match (
-            &self.const_expressions[left],
-            &self.const_expressions[right],
-        ) {
+        span: Span,
+    ) -> Result<Handle<Expression>, ConstantEvaluatorError> {
```

Comment:
GitHub won't let me put this comment on the right line, but:

The cases here for `BinaryOperator::Divide` and `Modulo` need to report divide by zero errors, not just flush them to zero.

Reply:
Resolved in https://github.com/gfx-rs/naga/pull/2309/commits/e139963fde2a18dad342a19b7d0933df320aba86